src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -614,7 +615,13 @@ public override int GetHashCode()
             {
                 if (_hashCode == 0)
                 {
-                    _hashCode = StringComparer.OrdinalIgnoreCase.GetHashCode(ToString());
+                    var hashCode = HashHelpers.Combine(HashHelpers.RandomSeed, unchecked((int)PrivateScopeId));
+                    for (int i = 0; i < NumberOfLabels; i += 2)
+                    {
+                        hashCode = HashHelpers.Combine(hashCode, _numbers[i] << sizeof(ushort) | _numbers[i + 1]);
```

Comment:
sizeof(ushort) is in bytes, not bits -- 8*sizeof(ushort)

Reply:
Of course it is :).
I had it at 16 originally. That's what I get for trying to be smart.